### PR TITLE
Feature/test gsa prep

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR adds a test for the GSA prep script, while also parameterising it for operation outside atlas-prod and calling bash functions via atlas-bash-util.

### CODE REVIEW

File: gsa/scripts/gsa_prepare_data.sh [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
+rm -rf $WORKDIR/*
 
 echo "About to assemble organism-experiments tuples"
-grep organism contrastdetails.tsv | grep -P '\torganism\t' | awk -F"\t" '{print $6"\t"$1}' | sort | uniq | sed 's| |_|g' > ${wDir}/exps.aux
-awk '$1 = tolower($1)' ${wDir}/exps.aux > ${wDir}/exps.aux.lc && mv ${wDir}/exps.aux.lc ${wDir}/exps.aux
-perl -pi -e 's|^oryza_sativa |oryza_sativa_japonica_group |g' ${wDir}/exps.aux
-perl -pi -e 's|^oryza_sativa_japonica |oryza_sativa_japonica_group |g' ${wDir}/exps.aux
+grep organism contrastdetails.tsv | grep -P '\torganism\t' | awk -F"\t" '{print $6"\t"$1}' | sort | uniq | sed 's| |_|g' > ${WORKDIR}/exps.aux
+awk '$1 = tolower($1)' ${WORKDIR}/exps.aux > ${WORKDIR}/exps.aux.lc && mv ${WORKDIR}/exps.aux.lc ${WORKDIR}/exps.aux
+perl -pi -e 's|^oryza_sativa |oryza_sativa_japonica_group |g' ${WORKDIR}/exps.aux
```
Comment (pcm32) [inline gsa/scripts/gsa_prepare_data.sh:25]: We should probably leave an open issue here on these dangling species transformations. We took a more centralised approach in bulk recalculations based on a shared YAML file that describe these mappings. Not sure if those directly apply here, but we should keep track maybe with an issue that this is here.

File: gsa/scripts/gsa_prepare_data.sh [hunk 2]
```diff
@@ -1,40 +1,43 @@
...
 echo "About to assemble tsv input lists for gsa_prepare_data.R" 
-for l in $(cat ${wDir}/exps.aux); do
+
+cat ${WORKDIR}/exps.aux | while read -r l; do
     o=`echo $l | awk '{print $1}'`
     e=`echo $l | awk '{print $2}'`
     ls $ATLAS_EXPS/$e/${e}*-analytics.tsv > /dev/null
     if [ $? -eq 0 ]; then 
        ls $ATLAS_EXPS/$e/$e-configuration.xml > /dev/null
        if [ $? -eq 0 ]; then
```
Comment (pcm32) [inline gsa/scripts/gsa_prepare_data.sh]: maybe shorter?
```suggestion
       if [ -f $ATLAS_EXPS/$e/$e-configuration.xml ]; then 
```

**Review by pcm32 — APPROVED** [review]
Great! Added a minor suggestion to simplify this.

File: gsa/scripts/gsa_prepare_data.sh (hunk 2 above)
Comment (pinin4fjords) [reply gsa/scripts/gsa_prepare_data.sh]: Thanks! I wasn't being very critical of the code.

File: gsa/scripts/gsa_prepare_data.sh (hunk 1 above)
Comment (pinin4fjords) [reply gsa/scripts/gsa_prepare_data.sh:25]: Yes, I do the same thing in single-cell. We should centralise that config as part of https://github.com/ebi-gene-expression-group/atlas-prod/pull/229

Comment (pinin4fjords) [reply gsa/scripts/gsa_prepare_data.sh:25]: Will make the issue

**Comment by pinin4fjords** [comment]
Thanks for review @pcm32

**Pull request merged**